### DISCUSSION TITLE
updated lifecycle hook V2 vs V3

### DISCUSSION THREAD

**Discussion by akamasAT** [livewire/livewire#8280] [1. Help]
Hello,

I am using a bootstrap theme. Rendering Livewire Components "destroys" some functionallity like tooltips.

In V2 i came up with a Trait that I added to my components, that worked fine

```
trait HasTooltipps
{
    public function renderedHasTooltipps()
    {
        $this->dispatch('renderTooltipps');
    }
}
```

listener:
```
Livewire.on('renderTooltipps', event => {
      $('[data-bs-toggle="tooltip"]').tooltip('dispose');
      $('[data-bs-toggle="tooltip"]').tooltip();
});
```

in V3 this is not working any more. The event is fireing and the listener working.
Has anyone an idea?

**Comment by gazzoy** [comment]
@akamasAT Have you tried `$wire.on` or listening events inside of  `livewire:init` as [documented](https://livewire.laravel.com/docs/events#listening-for-events)?

```php
@script
<script>
    $wire.on('renderTooltipps', () => {
        //
    });
</script>
@endscript
```

```php
<script>
    document.addEventListener('livewire:init', () => {
       Livewire.on('renderTooltipps', (event) => {
           //
       });
    });
</script>
```

**Reply by akamasAT** [reply]
thanks for your response. I use the "global js listener" as documented here: https://livewire.laravel.com/docs/events#listening-for-livewire-events-from-global-javascript

the listener is working fine

**Reply by gazzoy** [reply]
Cool, no worries.
